src/index.ts

```diff
@@ -129,3 +131,36 @@ function getCompilerOptions(tsConfigFilePath?: string): ts.CompilerOptions {
 
     return convertedCompilerOptions.options;
 }
+
+function writeFile(output: string, input: string): void {
+    // tslint:disable-next-line non-literal-fs-path
+    fs.writeFile(output, input, 'binary', (errNoException: NodeJS.ErrnoException | null): void => {
+        if (errNoException !== null) {
+            throw errNoException;
+        }
+
+        console.log('The file was saved!');
+    });
+}
+
+function requestImageFile(output: string, input: string, extension: string): void {
+    http.get({
+        host: 'www.plantuml.com',
+        path: `/plantuml/${extension}/${encode(input)}`
+    },       (res: http.IncomingMessage): void => {
+        let imagedata: string = '';
+        res.setEncoding('binary');
+
```

Comment:
I am pretty sure this can be simplified and you can `var stream = fs.createWriteStream(output);res.pipe(writeStream)` and avoid having to concatenate the image data into a string manually.

Reply:
Done, thanks